tests/unit/test_charm.py

```diff
@@ -470,6 +468,68 @@ def test_mongodb_relation_joined_peer_ready(
             else:
                 mongodb_reconfigure.assert_not_called()
 
+    @patch_network_get(private_address="1.1.1.1")
+    @patch("charm.MongodbOperatorCharm._single_mongo_replica")
+    @patch("charm.MongodbOperatorCharm._need_replica_set_reconfiguration", return_value=True)
+    @patch("mongod_helpers.MongoDB.reconfigure_replica_set")
+    @patch("mongod_helpers.MongoDB.check_replica_status")
+    def test_mongodb_relation_joined_check_status_failure(
+        self, check_replica_status, mongodb_reconfigure, need_reconfiguration, single_replica
+    ):
+        """Test failure in checking status results in waiting.
+
+        Tests the scenario that a failure to check the current statuses of replica set hosts
+        results in WaitingStatus and no attempt to reconfigure is made.
+        """
+        # preset values
+        self.harness.set_leader(True)
+        single_replica.return_value.is_ready.return_value = True
+
+        for exception in PYMONGO_EXCEPTIONS:
+            # simulate failure to check replica status
+            check_replica_status.side_effect = exception
+
+            # simulate 2nd MongoDB unit
+            rel = self.harness.charm.model.get_relation("mongodb")
+            key_values = {"private-address": "127.4.5.6"}
+            self.harness.add_relation_unit(rel.id, "mongodb/1")
+            self.harness.update_relation_data(rel.id, "mongodb/1", key_values)
+
+            # verify we go into waiting and don't reconfigure
+            self.assertEqual(
+                self.harness.charm.unit.status, WaitingStatus("waiting to reconfigure replica set")
+            )
+            mongodb_reconfigure.assert_not_called()
+
+    @patch_network_get(private_address="1.1.1.1")
+    @patch("charm.MongodbOperatorCharm._single_mongo_replica")
+    @patch("charm.MongodbOperatorCharm._need_replica_set_reconfiguration", return_value=True)
+    @patch("mongod_helpers.MongoDB.reconfigure_replica_set")
+    @patch("mongod_helpers.MongoDB.all_replicas_ready", return_value=False)
+    def test_mongodb_relation_joined_all_replicas_not_ready(
+        self, all_replicas_ready, mongodb_reconfigure, need_reconfiguration, single_replica
+    ):
+        """Tests that when current replcia set hosts are not ready that we go into waiting.
```

Comment:
This docstring is slightly clunky to read - consider using this: 
`"""Tests that we go into waiting when current ReplicaSet hosts are not ready"""`

Reply:
Awesome, thanks for this recommendation! :D